Parse/src/main/java/com/parse/ParseFile.java

```diff
@@ -134,12 +134,74 @@ public String url() {
    * successfully synced with the server.
    */
   /* package for tests */ byte[] data;
+  /* package for tests */ File file;
 
   /* package for tests */ final TaskQueue taskQueue = new TaskQueue();
   private Set<Task<?>.TaskCompletionSource> currentTasks = Collections.synchronizedSet(
       new HashSet<Task<?>.TaskCompletionSource>());
 
   /**
+   * Creates a new file from a file pointer, file name, and content type. Content type will be used
+   * instead of auto-detection by file extension.
+   *
+   * @param name
+   *          The file's name, ideally with extension. The file name must begin with an alphanumeric
+   *          character, and consist of alphanumeric characters, periods, spaces, underscores, or
+   *          dashes.
+   * @param file
+   *          The file.
+   * @param contentType
+   *          The file's content type.
+   */
+  public ParseFile(String name, File file, String contentType) {
```

Comment:
Is there any real reason we need to pass in `name` since we can get it from `File#getName()`?

/cc @nlutsenko since it looks like we have this constructor on iOS.